lib/pages/tickets/tickets.dart

```diff
@@ -0,0 +1,368 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+import 'package:tawkie/config/app_config.dart';
+import 'package:tawkie/pages/tickets/tickets_page.dart';
+import 'package:tawkie/utils/platform_infos.dart';
+import 'package:tawkie/utils/room_utils.dart';
+import 'package:tawkie/widgets/matrix.dart';
+
+import 'error_dialog.dart';
+import 'model/ticket_model.dart';
+
+class Tickets extends StatefulWidget {
+  const Tickets({super.key});
+
+  @override
+  TicketsController createState() => TicketsController();
+}
+
+class TicketsController extends State<Tickets> {
+  String userId = AppConfig.ticketsBotId;
+  List<Room> filteredRooms = [];
+  List<Ticket> tickets = [];
+  List<Ticket> filteredTickets = [];
+  bool loading = true;
+
+  bool stopProcess = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _initializeTickets();
+  }
+
+  @override
+  void dispose() {
+    stopProcess = true;
+    super.dispose();
+  }
+
+  Future<void> _initializeTickets() async {
+    await _getRoomsForUser();
+    for (var room in filteredRooms) {
+      await getTicketsFromRoom(room);
+    }
+
+    if (mounted) {
+      setState(() {
+        loading = false;
+        filteredTickets = tickets;
+      });
+    }
+  }
+
+  // Filter tickets by search text
+  void filterTickets(String searchText) {
+    setState(() {
+      if (searchText.isEmpty) {
+        filteredTickets = tickets;
+      } else {
+        filteredTickets = tickets
+            .where((ticket) =>
+                ticket.content.toLowerCase().contains(searchText.toLowerCase()))
+            .toList();
+      }
+    });
+  }
+
+  // Method to recover and filter rooms
+  Future<void> _getRoomsForUser() async {
+    List<Room> filteredRooms = await getTicketRooms();
+
+    if (mounted) {
+      setState(() {
+        this.filteredRooms = filteredRooms;
+      });
+    }
+
+    // For each filtered room, retrieve the tickets
+    for (var room in filteredRooms) {
+      await getTicketsFromRoom(room);
+    }
+  }
+
+  Future<void> _checkRoomParticipants(Room room) async {
+    // Always check participants' status, even if tickets have already been collected
+    List<User> participants = room.getParticipants();
+    String status = (participants.length == 2) ? 'open' : 'closed';
```

Comment:
in some cases other users could join the conversation. Please check for whether Honoroit is in the participants' list